client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -269,6 +269,7 @@ public void close()
             }
         }
         finally {
+            httpClient.connectionPool().evictAll();
```

Comment:
I would add `httpClient.dispatcher().executorService().shutdown()` before this line.

Reply:
Sure, added.

Reply:
This look wrong. The `httpClient.dispatcher()` is shared between different `Connections`.

See [here](https://github.com/trinodb/trino/blob/48e0bfa4fccf0ca789b6302185ef01865362d0a9/client/trino-jdbc/src/main/java/io/trino/jdbc/NonRegisteringTrinoDriver.java#L55).

It calls
```
  public Builder newBuilder() {
    return new Builder(this);
  }
```

and then

```
    Builder(OkHttpClient okHttpClient) {
      this.dispatcher = okHttpClient.dispatcher;
```


Please add test for that, closing one connection that does not prevent the other one from working.